shapes.js

```diff
@@ -0,0 +1,348 @@
+      myState.valid = false; // Something's dragging so we must redraw 
+    } else if (myState.enlarging == Shape.prototype.borderenum.BOTTOMRIGHT) {
+      myState.selection.w = mouse.x - myState.selection.x; 
+      myState.selection.h = mouse.y - myState.selection.y;  
+      myState.valid = false; // Something's dragging so we must redraw 
+    } else if (myState.enlarging == Shape.prototype.borderenum.BOTTOMLEFT) {
+      myState.selection.w += myState.selection.x - mouse.x; 
+      myState.selection.x = mouse.x;
+      myState.selection.h = mouse.y - myState.selection.y;   
+      myState.valid = false; // Something's dragging so we must redraw 
+    } else if (myState.enlarging == Shape.prototype.borderenum.TOPLEFT) {
+      myState.selection.w += myState.selection.x - mouse.x; 
+      myState.selection.x = mouse.x;
+      myState.selection.h += myState.selection.y - mouse.y; 
+      myState.selection.y = mouse.y;    
+      myState.valid = false; // Something's dragging so we must redraw 
+    } else if (myState.enlarging == Shape.prototype.borderenum.TOPRIGHT) {
+      myState.selection.h += myState.selection.y - mouse.y; 
+      myState.selection.y = mouse.y;    
+      myState.selection.w = mouse.x - myState.selection.x; 
+      myState.valid = false; // Something's dragging so we must redraw 
+    } else if (myState.dragging){
+      // We don't want to drag the object by its top-left corner, we want to drag it
+      // from where we clicked. Thats why we saved the offset and use it here
+      myState.selection.x = mouse.x - myState.dragoffx;
+      myState.selection.y = mouse.y - myState.dragoffy;   
+      myState.valid = false; // Something's dragging so we must redraw
+    }
+  }, true);
+  html.addEventListener("keypress", function(event) {
+    console.log(myState.selection);
+    if (event.keyCode == 32) {
+      if (myState.shapes.length > 0) {
+        myState.shapeNum = (myState.shapeNum + 1) % myState.shapes.length
+        myState.selection = myState.shapes[myState.shapeNum % myState.shapes.length];
+        console.log(myState.shapeNum);
+        myState.valid = false;
+      }
+    }
+  }, true);
+  html.addEventListener("keydown", function(event) {
+    if (myState.selection != null) {
+      if (event.keyCode == 38) {
+        myState.selection.y -= 1;
+      } else if (event.keyCode == 40) {
+        myState.selection.y += 1;
+      } else if (event.keyCode == 37) {
+        myState.selection.x -= 1;
+      } else if (event.keyCode == 39) {
+        myState.selection.x += 1;
+      } else if (event.keyCode == 68) {
+        myState.removeShape(myState.selection);
+        myState.selection = null;
+      }
+    } else if (event.keyCode == 37) {
+      if (myState.frame > 0) {
+        canvas.style.backgroundImage = frame_url(--(myState.frame));
+      }
+    } else if (event.keyCode == 39) {
+      canvas.style.backgroundImage = frame_url(++(myState.frame));
```

Comment:
This doesn't seem to work? 
Might be clearer to increment & decrement on a separate line